File: 1.15.c
#include <stdio.h>

/*
 *	rewrite of temperature conversion using functions
 */

#define LOWER 0
#define UPPER 300
#define STEP  20

float toCels(int fahr);
float toFahr(int cels);

int main()
{
	int fahr;

	for(fahr = LOWER; fahr <= UPPER; fahr = fahr + STEP)
		printf("%3d %6.1f\n", fahr, toCels(fahr));
}

float toCels(int fahr)
{
	return ((5.0/9.0)*(fahr-32));
}

float toFahr(int cels)
{
	return ((5.0/9.0)*(cels+32));
}

File: 1.15.py
#!/usr/bin/env python3

# rewrite the temperature converter with a function

lower = 0
upper = 300
step = 20

def to_fahr(cels):
    return (9/5)*(cels+32)

def to_cels(fahr):
    return (5/9)*(fahr-32)

fahr = lower
while fahr <= upper:
    print("%3.0f %6.1f" % (fahr, to_cels(fahr)))
    fahr += step
